docs/source/backend.rst

```diff
@@ -3,44 +3,88 @@
 torchaudio.backend
 ==================
 
+Overview
+~~~~~~~~
+
 :mod:`torchaudio.backend` module provides implementations for audio file I/O, using different backend libraries.
 To switch backend, use :py:func:`torchaudio.set_audio_backend`. To check the current backend use :py:func:`torchaudio.get_audio_backend`.
 
-.. warning::
-   Although ``sox`` backend is default for backward compatibility reason, it has a number of issues, therefore it is highly recommended to use ``sox_io`` backend instead. Note, however, that due to the interface refinement, functions defined in ``sox`` backend and those defined in ``sox_io`` backend do not have the same signatures.
+There are currently four implementations available.
 
-.. note::
-   Instead of calling functions in :mod:`torchaudio.backend` directly, please use ``torchaudio.info``, ``torhcaudio.load``, ``torchaudio.load_wav`` and ``torchaudio.save`` with proper backend set with :func:`torchaudio.get_audio_backend`.
+    * ``"sox"`` (deprecated)
+    * ``"sox_io"``
+    * ``"soundfile"``, legacy interface (deprecated)
+    * ``"soundfile"``, new interface
+
+``"sox"`` backend is the original backend which is built on ``libsox``. This backend requires C++ extension module and is not available on Windows system. This backend is currently the default backend when available, but is being deprecated. Please use ``"sox_io"`` backend.
 
-There are currently three implementations available.
+``"sox_io"`` backend is the new backend which is built on ``libsox``, and bound with ``TorchScript``. This backend requires C++ extension module and is not available on Windows system. Function calls to this backend are TorchScript-able. This backend will become the default backend in ``"0.8.0"`` release.
```

Comment:
You use the word backend 4 times here

Reply:
"and bound with ``TorchScript``" doesn't mean much to most users I'd say. Instead write that this implies.

Reply:
"This backend requires C++ extension module" - Is this something I'd need to install or worry about in most cases? Or is it only something to worry about on Windows? If I'm on Windows, what can I do? (add this to the intro). You probably want to have a platform-specific suggestion or guidance in the intro (might simply be Windows vs. not-Windows) and also again whether the user needs to worry or more specifically when the user needs to worry (e.g. mp3 support).

Bonus points for mentioning we're working on resolving some of these issues, e.g. mp3 work in progress for Windows etc., and a link to the corresponding issue

Reply:
Addressed all the suggestions. I am hesitant to mention the new I/O work as it's very early phase and we do not know when it will be available.